autoscale/src/main/java/com/sequenceiq/periscope/monitor/handler/FreeIpaCommunicator.java

```diff
@@ -33,17 +42,36 @@ public FreeIpaCommunicator(UserV1Endpoint userV1Endpoint) {
     public SyncOperationStatus synchronizeAllUsers(SynchronizeAllUsersRequest request) {
         LOGGER.info("Invoking freeIpa user sync request: {}", request);
         String envCrn = request.getEnvironments().iterator().next();
-        SyncOperationStatus lastSyncStatus = ThreadBasedUserCrnProvider.doAsInternalActor(
-                internalCrnGeneratorFactory.autoscale().getInternalCrnForServiceAsString(),
-                () -> userV1Endpoint.getLastSyncOperationStatus(envCrn));
-        if (SynchronizationStatus.RUNNING.equals(lastSyncStatus.getStatus())) {
-            LOGGER.info("There is a user sync operation already running for environment: {} with operationId: {}, " +
-                    "skipping request to trigger another user sync", envCrn, lastSyncStatus.getOperationId());
-            return lastSyncStatus;
-        } else {
-            return ThreadBasedUserCrnProvider.doAsInternalActor(
+        try {
+            SyncOperationStatus lastSyncStatus = ThreadBasedUserCrnProvider.doAsInternalActor(
+                    internalCrnGeneratorFactory.autoscale().getInternalCrnForServiceAsString(),
+                    () -> userV1Endpoint.getLastSyncOperationStatus(envCrn));
+            if (SynchronizationStatus.RUNNING.equals(lastSyncStatus.getStatus())) {
+                LOGGER.info("There is a user sync operation already running for environment: {} with operationId: {}, " +
+                        "skipping request to trigger another user sync", envCrn, lastSyncStatus.getOperationId());
+                return lastSyncStatus;
+            } else {
+                return invokeFreeipaUserSyncAndHandleException(request, envCrn);
+            }
+        } catch (Exception ex) {
+            String message = format("Encountered error when communicating with IPA service for environment: %s", envCrn);
+            LOGGER.error(message, ex);
+            throw new RuntimeException(message, ex);
+        }
+    }
+
+    private SyncOperationStatus invokeFreeipaUserSyncAndHandleException(SynchronizeAllUsersRequest request, String envCrn) {
+        try {
+            SyncOperationStatus status = ThreadBasedUserCrnProvider.doAsInternalActor(
                     internalCrnGeneratorFactory.autoscale().getInternalCrnForServiceAsString(),
                     () -> userV1Endpoint.synchronizeAllUsers(request));
+            metricService.incrementMetricCounter(IPA_USER_SYNC_INVOCATION, MetricTag.TENANT.name(), request.getAccountId());
+            return status;
+        } catch (Exception ex) {
+            LOGGER.error("Failed to synchronize users to IPA for environment: {}", envCrn, ex);
+            metricService.incrementMetricCounter(IPA_USER_SYNC_FAILED, MetricTag.TENANT.name(), request.getAccountId());
+
+            throw new RuntimeException(format("User sync failed for environment %s", envCrn), ex);
```

Comment:
The invoked method is not set to throw checked exceptions. To avoid unnecessary nesting of Exception (and propagating the original Exception) - this should check if the Exception is a RuntimeException - if it is, cast and rethrow, if not - then convert it to an explicit RuntimeException.